pkg/controller/registry/resolver/runtime_constraints/provider.go

```diff
@@ -0,0 +1,80 @@
+package runtime_constraints
+
+import (
+	"encoding/json"
+	"io/ioutil"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/registry/resolver/cache"
+	"github.com/operator-framework/operator-registry/pkg/registry"
+	"github.com/pkg/errors"
+)
+
+const (
+	maxRuntimeConstraints = 10
+)
+
+type RuntimeConstraintsProvider struct {
+	runtimeConstraints []cache.Predicate
+}
+
+func (p *RuntimeConstraintsProvider) GetRuntimeConstraints() []cache.Predicate {
+	return p.runtimeConstraints
+}
+
+func New(runtimeConstraints []cache.Predicate) *RuntimeConstraintsProvider {
+	return &RuntimeConstraintsProvider{
+		runtimeConstraints: runtimeConstraints,
+	}
+}
+
+func NewFromFile(runtimeConstraintsFilePath string) (*RuntimeConstraintsProvider, error) {
+	propertiesFile, err := readRuntimeConstraintsYaml(runtimeConstraintsFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Using package type to test with
+	// We may only want to allow the generic constraint types once they are readym
+	var constraints = make([]cache.Predicate, 0)
+	for _, property := range propertiesFile.Properties {
```

Comment:
I understand this will be substituted with `olm.constraint` type later which sounds good. If you are interested in testing this on a live cluster with CatalogSources that contain bundles with `MaxOCPVersion` properties, you can add some hardcoded stuff to handle `MaxOCPVersion` dependency which is similar to package dependency. Just a suggestion.

Reply:
That's not a bad idea! Can i use CRC for this? Or should I use the slack bot?